### PR TITLE
Update to roslyn analyzers 3.3.5-beta1.23180.2

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by ToddGrun — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs [hunk 1]
```diff
@@ -5540,7 +5540,7 @@ void M()
 
             compilation.VerifyDiagnostics();
 
-            foreach (var declarationSyntax in tree.GetCompilationUnitRoot().DescendantTokens().OfType<VariableDeclarationSyntax>().Where(syntax => syntax.GetFirstToken().IsKind(SyntaxKind.VoidKeyword)))
+            foreach (var declarationSyntax in tree.GetCompilationUnitRoot().DescendantNodes().OfType<VariableDeclarationSyntax>().Where(syntax => syntax.GetFirstToken().IsKind(SyntaxKind.VoidKeyword)))
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs:5543]: interesting.  seems like the old code could never have worked before...

Comment (sharwell) [reply src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs:5543]: The old code contained assertions that never executed. This is one example of the type of error that code coverage can help prevent, provided you include code coverage data for test assemblies.
https://github.com/coverlet-coverage/coverlet/issues/618

Comment (sharwell) [reply src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs:5543]: /cc @StingyJack @PureKrome for a real-world example of a situation that would have benefitted from code coverage on test code

Comment (MarcoRossignoli) [reply src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs:5543]: thanks @sharwell for the real-case.

**Pull request merged**